src/main/java/org/wso2/carbon/salesforce/connector/SalesforceSOAPLoginHandler.java

```diff
@@ -0,0 +1,131 @@
+                            "xmlns:urn=\"urn:partner.soap.sforce.com\">" +
+                            "<soapenv:Body>" +
+                            "<urn:login>" +
+                            "<urn:username>" + username + "</urn:username>" +
+                            "<urn:password>" + password + "</urn:password>" +
+                            "</urn:login>" +
+                            "</soapenv:Body>" +
+                            "</soapenv:Envelope>";
+
+            HttpPost post = new HttpPost(loginUrl);
+            post.setHeader("Content-Type", "text/xml; charset=UTF-8");
+            post.setHeader("SOAPAction", "urn:partner.soap.sforce.com/Soap/loginRequest");
+            post.setEntity(new StringEntity(soapBody, StandardCharsets.UTF_8));
+
+            try (CloseableHttpClient client = HttpClients.createDefault();
+                 CloseableHttpResponse response = client.execute(post)) {
+
+                int statusCode = response.getStatusLine().getStatusCode();
+
+                HttpEntity entity = response.getEntity();
+                if (entity == null) {
+                    throw new ConnectException("Empty SOAP login response");
+                }
+                String responseXml = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+                DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+                Document doc = db.parse(
+                        new ByteArrayInputStream(responseXml.getBytes(StandardCharsets.UTF_8)));
+
+                XPath xp = XPathFactory.newInstance().newXPath();
+
+                Node faultNode = (Node) xp.evaluate(
+                        "/*[local-name()='Envelope']/*[local-name()='Body']/*[local-name()='Fault']"
+                                + "/*[local-name()='faultstring']",
+                        doc, XPathConstants.NODE);
+
+                if (faultNode != null) {
+                    throw new ConnectException("Salesforce login failed: "
+                            + faultNode.getTextContent());
+                }
+
+                String sessionId = xp.evaluate("string(//*[local-name()='sessionId'])", doc);
+                String serverUrl = xp.evaluate("string(//*[local-name()='serverUrl'])", doc);
+
+                if (sessionId.isEmpty() || serverUrl.isEmpty()) {
+                    throw new ConnectException("sessionId or serverUrl not found in login response");
+                }
+
+                System.out.println("Parsed values → sessionId=" + sessionId
+                        + ", serverUrl=" + serverUrl);
+
+                messageContext.setProperty("salesforce.sessionId", sessionId);
+                messageContext.setProperty("salesforce.serviceUrl", serverUrl);
+                messageContext.setProperty("salesforce.login.done", "true");
+
+                System.out.println("Saved properties on MessageContext");
```

Comment:
Shall we use proper log statements instead of system.out.println? Please check in other places as well.